Source/WebKit/UIProcess/Extensions/WebExtension.cpp

```diff
@@ -245,6 +245,65 @@ void WebExtension::populateWebAccessibleResourcesIfNeeded()
         parseWebAccessibleResourcesVersion2();
 }
 
+URL WebExtension::resourceFileURLForPath(const String& originalPath)
+{
+    ASSERT(originalPath);
+
+    String path = originalPath;
+
+    if (path.startsWith('/'))
+        path = path.substring(1);
+
+    if (!path.length() || m_resourceBaseURL.isNull())
+        return { };
+
+    auto resourceURL = URL(m_resourceBaseURL, path);
+
+    // Don't allow escaping the base URL with "../".
+    if (!resourceURL.string().startsWith(m_resourceBaseURL.string())) {
```

Comment:
I'm skeptical that this does the same thing as before. The NSURL path resolved all the `../` and `./` in the path before doing this prefix check. Is that being done here?

Reply:
Specifically the `URLByStandardizingPath` part of the old code. Not doing that would cause a security issue here.

Reply:
Looking at it again, along with the definitions for WTF::Url, I'm not sure if there's a way for WTF::Url to do that (and also, I'd imagine path resolving acts different on different platforms). Unless there's something in WTF I'm missing, I wonder if it would be better to take an NSUrl and get the URLByStandardisingPath on Cocoa, and do something different on other platforms

Reply:
Actually, I may have missed something. There is a `fileSystemPath()` function in URL, I wonder if that would resolve first?

Reply:
I think `fileSystemPath()` might still have the `../`. You should test to verify if it does or not, an automation test would be good here too.

Reply:
Yeah that's fine, I can investigate doing something different for glib later